### ISSUE TITLE
"List was empty at reduce"

### ISSUE THREAD

**Issue by tpearson1** [stkb/Rewrap#222]
After pressing `Alt+Q`, I experienced this. Pressing `Alt+Q` again (and again) triggers the same error.

```
Error: List was empty
  at reduce (c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\compiled\fable-library.2.10.1\List.js:810:11)
  at min (c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\compiled\fable-library.2.10.1\List.js:1009:10)
  at List.fencedCodeBlock [as head] (c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\compiled\Parsing.Markdown.js:73:30)
  at tryMany (c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\compiled\Parsing.Core.js:66:34)	
  at allParsers (c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\compiled\Parsing.Markdown.js:280:46)	
  at loop$$1 (c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\compiled\Parsing.Core.js:126:29)	
  at c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\compiled\fable-library.2.10.1\Util.js:815:24	
  at c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\compiled\Parsing.Markdown.js:289:7	
  at Object.rewrap (c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\compiled\Main.js:78:34)	
  at doWrap (c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\Extension.js:159:29)	
  at rewrapCommentCommand (c:\Users\<me>\.vscode\extensions\stkb.rewrap-1.12.0\vscode\Extension.js:28:9)	
  at c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:911:271	
  at t.ExtHostTextEditor.edit (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:586:525)	
  at c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:911:260	
  at _executeContributedCommand (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:631:163)	
  at _.$executeContributedCommand (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:631:538)	
  at m._doInvokeHandler (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:801:683)	
  at m._invokeHandler (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:801:375)	
  at m._receiveRequest (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:800:14)	
  at m._receiveOneMessage (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:798:832)	
  at c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:797:34	
  at l.fire (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:45:801)	
  at v.fire (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:235:849)	
  at c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:975:744	
  at l.fire (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:45:801)	
  at v.fire (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:235:849)	
  at t.PersistentProtocol._receiveMessage (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:240:276)	
  at c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:237:357	
  at l.fire (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:45:801)	
  at p.acceptChunk (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:233:393)	
  at c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:232:753	
  at Socket.t (c:\Users\<me>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:241:826)	
  at Socket.emit (events.js:203:13)	
  at addChunk (_stream_readable.js:295:12)	
  at readableAddChunk (_stream_readable.js:276:11)	
  at Socket.Readable.push (_stream_readable.js:210:10)	
  at Pipe.onStreamRead (internal/stream_base_commons.js:166:17)
```

**Comment by stkb** [comment]
Hi, can you give a sample of text that causes this error? (It looks like a markdown file with a fenced code block ` ```...``` `)

**Comment by tpearson1** [comment]
Yep, if I press Alt+Q from a new VS Code window with the cursor at the location as seen in the image, the error can be reproduced.

![image](https://user-images.githubusercontent.com/16149168/87635561-63667380-c793-11ea-9152-7c408b681784.png)

**Comment by stkb** [comment]
Thanks. Should now be fixed in v1.13.0
